includes/class-wc-payments.php

```diff
@@ -418,7 +419,20 @@ public static function add_plugin_links( $links ) {
 	 */
 	public static function register_gateway( $gateways ) {
 		$gateways[] = self::$card_gateway;
-		$gateways[] = self::$sepa_gateway;
+
+		if ( false === WC_Payments_Features::is_grouped_settings_enabled() ) {
+			return $gateways;
+		}
+
+		// adding the sepa/giropay/etc gateways after the feature flag, to ensure that there are no side effects.
+		// these lines ensure that the payment method is added to the array only when we're _not_ on the WC settings page.
+		if (
+			( empty( $_GET['page'] ) && empty( $_GET['tab'] ) ) // phpcs:ignore
+			||
+			( ! empty( $_GET['page'] ) && ! empty( $_GET['tab'] ) && 'wc-settings' === $_GET['page'] && 'checkout' === $_GET['tab'] && ! empty( $_GET['section'] ) ) // phpcs:ignore
```

Comment:
I originally switched to using the css to hide the gateway because this check broke down when I was playing with making WCPay setting take over the main section. I thought we should try to find something more robust. But of course the sorting being broken is something we can't have. I'll investigate if there's a more elegant solution. Probably this is tho only way to go and will need adjusted if we take over the settings page.

Reply:
Yeah, I have https://github.com/Automattic/woocommerce-payments/pull/1580 (in draft) where I'm going to try and have WCPay take over the default view for the payments section - it's working ok (without these changes), but I'm planning to integrate them and test everything in there